### ISSUE TITLE
Censored needs a special multiplication overload

### ISSUE THREAD

**Issue by sethaxen** [JuliaStats/Distributions.jl#1500]
Because the density at the bounds of `Censored` are discrete probabilities, they don't transform in the same way as the density between the bounds when the uncensored distribution is continuous. We can fix this with an overload like this:
```julia
*(d::Censored, σ::Real) = censored(d.uncensored * σ, d.lower * σ, d.upper * σ)
*(d::LeftCensored, σ::Real) = censored(d.uncensored * σ, d.lower * σ, nothing)
*(d::RightCensored, σ::Real) = censored(d.uncensored * σ, nothing, d.upper * σ)
```

More generally, this is something to keep in mind when considering other distributions with atomics.

**Comment by mschauer** [comment]
Connected to #1483, there is a latent need for being able to dispatch on distributions which have merely uncountable support and those which are absolute continuous.

**Comment by devmotion** [comment]
Indeed. I guess, however, here we would want the definitions suggested by @sethaxen in any case, even if we could dispatch on such traits.

**Comment by sethaxen** [comment]
> Connected to #1483, there is a latent need for being able to dispatch on distributions which have merely uncountable support and those which are absolute continuous.

I'm not certain how that would help in this case. Because we need the `log(σ)` for the continuous mixture component but need to not have it for the discrete components. If we have generic mixtures of discrete and continuous components, where a point can be any or all of the components, then even evaluation at a single pojnt might need the adjustment for some components and not others.
